x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,2777 @@
+{
+  name: "My prebuilt rule",
+  rule_source: {
+    type: "external",
+    is_customized: false,
+  }
+} 
+</pre>
+      </td>
+    <tr>
+  <tr>
+    <td>
+      Not installed
+    </td>
+    <td>
+<pre>
+{
+  name: "My prebuilt rule",
+} 
+</pre>
+    </td>
+    <td>Yes</td>
+    <td>
+<pre>
+{
+  name: "My prebuilt rule",
+} 
+</pre>
+    </td>
+    <td>N/A</td>
+    <td>
+<pre>
+{
+  name: "My prebuilt rule",
+  rule_source: {
+    type: "external",
+    is_customized: false,
+  }
+} 
+</pre>
+      </td>
+    <tr>
+  <tr>
+    <td>
+      Not installed
+    </td>
+    <td>
+<pre>
+{
+  name: "My prebuilt rule",
+} 
+</pre>
+    </td>
+    <td>Yes</td>
+    <td>
+<pre>
+{
+  name: "My custom prebuilt rule",
+} 
+</pre>
+    </td>
+    <td>N/A</td>
+    <td>
+<pre>
+{
+  name: "My custom prebuilt rule",
+  rule_source: {
+    type: "external",
+    is_customized: true,
+  }
+} 
+</pre>
+      </td>
+    <tr>
+  <tr>
+    <td>
+<pre>
+{
+  name: "My prebuilt rule",
+  rule_source: {
+    type: "external",
+    is_customized: false,
+  }
+} 
+</pre>
+    </td>
+    <td>
+<pre>
+{
+  name: "My prebuilt rule",
+} 
+</pre>
+    </td>
+    <td>Yes</td>
+    <td>
+<pre>
+{
+  name: "My custom prebuilt rule",
+} 
+</pre>
+    </td>
+    <td><b>No</b></td>
+    <td>
+    Rule import rejected<br> because of rule_id match and overwrite flag is false
+      </td>
+    <tr>
+  <tr>
+    <td>
+<pre>
+{
+  name: "My prebuilt rule",
+  rule_source: {
+    type: "external",
+    is_customized: false,
+  }
+} 
+</pre>
+    </td>
+    <td>
+<pre>
+{
+  name: "My prebuilt rule",
+} 
+</pre>
+    </td>
+    <td>Yes</td>
+    <td>
+<pre>
+{
+  name: "My custom prebuilt rule",
+} 
+</pre>
+    </td>
+    <td>Yes</td>
+    <td>
+<pre>
+{
+  name: "My custom prebuilt rule",
+  rule_source: {
+    type: "external",
+    is_customized: true,
+  }
+} 
+</pre>
+      </td>
+    <tr>
+  <tr>
+    <td>
+<pre>
+{
+  name: "My custom prebuilt rule",
+  rule_source: {
+    type: "external",
+    is_customized: true,
+  }
+} 
+</pre>
+    </td>
+    <td>
+<pre>
+{
+  name: "My prebuilt rule",
+} 
+</pre>
+    </td>
+    <td>Yes</td>
+    <td>
+<pre>
+{
+  name: "My prebuilt rule",
+} 
+</pre>
+    </td>
+    <td>Yes</td>
+    <td>
+<pre>
+{
+  name: "My prebuilt rule",
+  rule_source: {
+    type: "external",
+    is_customized: false,
+  }
+} 
+</pre>
+      </td>
+    <tr>
+  </tbody>
+</table>
+
+
+### Handling the `version` parameter
+
+In the code above, we made an additional check: if the rule to be imported is a prebuilt rule, then the `version` field will be required as part of the payload as well. Attempting to import a prebuilt rule without it will result in an error and the importing of that rule will be skipped. The `version` is necessary to be able to provide updates to the rule when a new version is available in the Prebuilt Rules package. Without it we don't know if the installed version is newer or older than the update being offered by the package.
```

Comment:
Ok, so another important note. The `version` field should be required only for importing prebuilt rules. We should preserve the current behavior where users are able to import custom rules without specifying their versions.

In the current `RuleToImport` schema the `version` field comes from `BaseCreateProps` where it's defined as an optional field:

```
export const RuleToImport = BaseCreateProps.and(TypeSpecificCreateProps).and(
  ResponseFields.partial().extend({
    rule_id: RuleSignatureId,
    immutable: z.literal(false).default(false),
  })
);
```

I think this schema should stay the same with regards to the `version` field. We shouldn't override it and make the field required in `RuleToImport`. Otherwise, it would be a breaking change.

The rule import endpoint should:

- for custom rules being imported, if `version` is not specified, set it to 1
- for prebuilt rules being imported, if `version` is not specified, throw an error

Let's update this paragraph and the "Rule Import request schema" section accordingly.

Reply:
I updated the `#### Rule Import request schema` section to details this: 

- only `rule_id` is additionally required in the endpoint, `immutable` is not required anymore.
- whether the rule is prebuilt or not will be calculated based on the `rule_id` (searching in the `security-rule` SO for matches) (EDGE CASE TO RESOLVE: What if the package is not installed?)
- treatment of `version` in imported rules with the logic described above.